code/game/modifiers/modifiers.dm

```diff
@@ -0,0 +1,446 @@
+		return D.handle_registration(override)
+	else
+		return null//The modifier must have failed creation and deleted itself
+
+
+/datum/modifier
+//Config
+	var/check_interval = 300//How often, in deciseconds, we will recheck the validity
+	var/atom/target = null
+	var/atom/source = null
+	var/modifier_type = 0
+	var/source_data = 0
+	var/strength = 0
+	var/duration = null
+
+	//A list of equip slots which are considered 'worn'.
+	//For equipment modifier type to be valid, the source object must be in a mob's contents
+	//and equipped to one of these whitelisted slots
+	//This list can be overridden if you want a custom slot whitelist
+	var/list/valid_equipment_slots = list(slot_back, slot_wear_mask, slot_handcuffed, slot_belt, \
+	slot_wear_id, slot_l_ear, slot_glasses, slot_gloves, slot_head, slot_shoes, slot_wear_suit, \
+	slot_w_uniform,slot_legcuffed, slot_r_ear, slot_legs, slot_tie)
+
+
+
+//Operating Vars
+	var/active = 0//Whether or not the effects are currently applied
+	var/next_check = 0
+
+
+//If creation of a modifier is successful, it will return a reference to itself
+//If creation fails for any reason, it will return null as well as giving some debug output
+/datum/modifier/New(var/atom/_target, var/_modifier_type, var/_source = null, var/_source_data = 0, var/_strength = 0, var/_duration = 0, var/_check_interval = 0)
+	..()
+	target = _target
+	modifier_type = _modifier_type
+	source = _source
+	source_data = _source_data
+	strength = _strength
+	if (_duration)
+		duration = _duration
+
+	if (_check_interval)
+		check_interval = _check_interval
+
+	if (!target || !modifier_type)
+		return invalid_creation("No target and/or no modifier type was submitted")
+
+	switch (modifier_type)
+		if (MODIFIER_EQUIPMENT)
+			if (!istype(target, /mob))
+				return invalid_creation("Equipment type requires a mob target")
+
+			if (!source || !istype(source, /obj))
+				return invalid_creation("Equipment type requires an object source")
+
+			//TODO: Port equip slot var
+		if (MODIFIER_ITEM)
+			if (!source || !istype(source, /obj))
+				return invalid_creation("Item type requires a source")
+
+		if (MODIFIER_REAGENT)
+			if (!istype(target, /mob) || !istype(source, /datum/reagent))
+				return invalid_creation("Reagent type requires a mob target and a reagent source")
+
+		if (MODIFIER_AURA)
+			if (!source || !istype(source, /atom))
+				return invalid_creation("Aura type requires an atom source")
+
+		if (MODIFIER_TIMED)
+			if (!duration || duration <= 0)
+				return invalid_creation("Timed type requires a duration")
+		if (MODIFIER_CUSTOM)
+			//No code here, just to prevent else
+		else
+			return invalid_creation("Invalid or unrecognised modifier type")//Not a valid modifier type.
+	return 1
+
+
+/datum/modifier/proc/handle_registration(var/override = 0)
+	var/datum/modifier/existing = null
+	for (var/datum/modifier/D in target.modifiers)
+		if (D.type == type)
+			existing = D
+	if (!existing)
+		processing_modifiers |= src
```

Comment:
`|=` operator is slow. It goes through the entire list, looking for `src` in there, and if it doesn't find it, it adds it. Think you can write it to be safe enough to use the cheaper `+=` operator?